### ISSUE TITLE
Function to sort vector based on length of content

### ISSUE THREAD

**Issue by oobianom** [oobianom/quickcode#26]
Hey Brice, do you know if such a function already exists. Below is a simple example to explain plainly what I am looking for --

vector1 <- c("acs","tt","jdssr","h","grab")

lensort(vector1,"asc")

result: c("h","tt","acs","grab","jdssr")

lensort(vector1,"desc")

result: c("jdssr","grab","acs","tt","h")

**Comment by brichard1638** [comment]
The closest function I could find in R that duplicated the functionality you outlined can be found in the **mde** package. It is a function called 'sort_by_missingness.' This function returns a listing of sorted, columnated field names based on the total number of NAs (missing values) per field name. However, it does not reflect the functionality you described.

I think the function idea you suggested is a good one and should be included in the **quickcode** package.

Per your function requirements, I have developed a baseline version of this function for your review and consideration. I call this function **varLengthSort**. The functional framework is provided below:

`varLengthSort <- function(vec, asc = TRUE) {`
  `# Remove NA values`
  `vec <- vec[!is.na(vec)]`
  
  `# Compute lengths of elements`
  `lengths <- nchar(vec)`
  
  `# Sort indices based on lengths`
  `sorted_indices <- order(lengths, decreasing = !asc)`
  
  `# Return sorted vector`
  `sorted_vec <- vec[sorted_indices]`
 ` return(sorted_vec)`
`}`

**EXAMPLE:**
`x = c("acs","tt","jdssr","h","grab")`
`y = varLengthSort(vec = x)`
`z = varLengthSort(vec = x, asc = FALSE)`

I have tested the foregoing function at a moderate level, and believe it returns correctly defined vector elements, sorted both in an ascending or descending manner, depending on how the **asc** parameter is applied.

**Comment by oobianom** [comment]
Brice, thanks for looking into this. I actually installed the mde package and tested that function. Its quite unique and useful. But you are right, it is different from what I am proposing here. That said, the functional framework you have provided is fantastic! Thanks.  I just added it to the package. However, for the function name, I'd like us to use  "sort.length" instead as it is easier to find by the user since it follows the same naming convection as the other sort functions in the base R package.

![image](https://github.com/oobianom/quickcode/assets/9087578/5584fa3d-fade-4475-b1fd-31ffe10223af)
